### PR TITLE
Modernize bundling

### PR DESCRIPTION
**Pull request by sgtsquiggs**
* Use rollup for bundling
* Produce `lib` output via babel
* Integrate prettier into eslint
* Upgrade out of date dev dependencies

### CODE REVIEW

**Comment by z3bi** [comment]
Awesome! But let's keep the tests in the `test` directory. We probably also want to update the example.html to pull in the new correct file.

Also I'm wondering if we should commit the `lib` directory or generate it as a build artifact?

**Comment by z3bi** [comment]
Looking at other libraries like https://github.com/chartjs/Chart.js they don't commit their `dist` directory. This will also allow us to remove the check in the build script.

**Comment by sgtsquiggs** [comment]
> Awesome! But let's keep the tests in the `test` directory.

I’ve always kept them together but also my projects are very large. Keeping tests next to the code being tested is easier to work with when your repo is large. Fine to separate here.

> We probably also want to update the example.html to pull in the new correct file.

If we aren’t committing the end result (lib) then example won’t have anything to reference. Could create an example with its own package.json that would achieve the same end result.

> 
> Also I'm wondering if we should commit the `lib` directory or generate it as a build artifact?

We don’t need to commit it - it’ll be in the npm package and that’s good enough.

**Review by z3bi — APPROVED** [review]

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -28,9 +28,6 @@ jobs:
       - run: npm install -g npm@7 # update to npm 7
```
Comment (abumalick) [inline .github/workflows/node.js.yml:28]: You can install last npm version as you droped nodejs 10.we don't need to pin to 7 anymore

Comment (sgtsquiggs) [reply .github/workflows/node.js.yml:28]: ah true

**Pull request merged**